### PR TITLE
Feat/candle/initialize

### PR DESCRIPTION
**Pull request by louisfd**
I made the skeleton for the new [Candle](https://github.com/huggingface/candle) backend. 
_Add_ operation, along with basics such as from_data, is done. 

Next I will fill every mandatory operation 😄

### CODE REVIEW

File: burn-candle/src/backend.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+pub struct CandleBackend<F: FloatCandleElement, I: IntCandleElement> {
+    _float: PhantomData<F>,
+    _int: PhantomData<I>,
+}
+
+impl<F: FloatCandleElement, I: IntCandleElement> Backend for CandleBackend<F, I> {
+    type Device = CandleDevice;
+
+    type FullPrecisionBackend = CandleBackend<Self::FullPrecisionElem, Self::IntElem>;
+    type FullPrecisionElem = f64;
```
Comment (nathanielsimard) [inline burn-candle/src/backend.rs:60]: Full precision is normally f32, f64 is double precision.

**Review by nathanielsimard — APPROVED** [review]
A tiny change, but approve in advance.

**Review by nathanielsimard — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Wow, super impressive to see this going up that quickly! Thanks for putting this up! Definitely let us know if there is anything that we can do on the candle side to help or if you have any sort of feedback. Fwiw candle has a lot to be optimized yet, e.g. quantized tensors got a speedup of ~5x recently, accelerate on mac was just added - also a speedup of up to 5x, and there is a ton more to come in the next few weeks. It definetly helps if you report things being slow or even broken as we could prioritize these more heavily!

**Comment by nathanielsimard** [comment]
@LaurentMazare, we have a backend-agnostic set of tests that we intend to execute using Candle (https://github.com/burn-rs/burn/tree/main/burn-tensor/src/tests). Furthermore, we will be comparing the performance of the Candle backend with other backends, including `tch`, using the benchmarking tool we've developed (https://github.com/burn-rs/burn-benches).

If we face any issues or bugs, we'll let you know :)

**Comment by LaurentMazare** [comment]
Great, just to mention that for performance you may want to use the following features:
- `mkl` on intel platforms.
- `accelerate` on macOS.
- `cudnn` on nvidia GPUs - and `flash-attn` when using transformers architectures.
Even with that, I would expect candle to be quite crushed by tch at the moment but certainly looking forward to seeing some numbers (we also have to set up some benchmarking monitoring on our side). Hopefully this will get better over time (and the easiness of deployment of candle compared to tch/libtorch will still make it a good option).

**Pull request merged**